File: jobtracker/src/main/java/com/example/jobtracker/Service/GoogleMeetService.java
package com.example.jobtracker.Service;

import java.util.List;

import com.example.jobtracker.Dto.MeetingRequest;
import com.example.jobtracker.Dto.MeetingResponse;

public interface GoogleMeetService {
    MeetingResponse createMeeting(MeetingRequest request);
    List<MeetingResponse> getAllMeetings();
    MeetingResponse getMeetingById(Long id);
    MeetingResponse updateMeet(Long id,MeetingRequest meet);
    void deleteMeet(Long id);   
}

File: src/app/services/google-meet.service.ts
import { HttpBackend, HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Meeting,MeetingRequest } from "../models/meeting.model";



@Injectable({
    providedIn:'root'
})
export class GoogleMeetService{

    private apiurl='http://localhost:8081/api/v1/google-meet'

    constructor(private http:HttpClient){}

    createMeeting (request:MeetingRequest){
        return this.http.post<Meeting>(`${this.apiurl}/create`,request);
    }

    getAllMeetings(){
        return this.http.get<Meeting[]>(`${this.apiurl}/all`);
    }

    getMeetingById(id:number){
        return this.http.get<Meeting>(`${this.apiurl}/${id}`);
    }

    updateMeeting(id:number,request:MeetingRequest){
        return this.http.put<Meeting>(`${this.apiurl}/updateMeet/${id}`,request);
    }

    deleteMeeting(id:number){
        return this.http.delete(`${this.apiurl}/deleteMeet/${id}`);
    }
}